repository_service_tuf/cli/admin/metadata.py

```diff
@@ -587,3 +605,117 @@ def update(
     else:
         # There are no changes made to the root metadata file.
         console.print("\nNo file will be generated as no changes were made\n")
+
+
+def _get_pending_signatures(
+    settings: Any, api_url: Optional[str]
+) -> Dict[str, Any]:
+    if settings.AUTH is False and api_url is None:
+        api_url = prompt.Prompt.ask("\n[cyan]API[/] URL address")
+        settings.SERVER = api_url
+
+    headers = get_headers(settings)
+    response = request_server(
+        settings.SERVER, URL.metadata_sign.value, Methods.get, headers=headers
+    )
+    if response.status_code != 200:
+        raise click.ClickException(
+            f"Failed to retrieve the sign data. Error: {response.text}"
+        )
+
+    response_data: Dict[str, Any] = response.json().get("data")
+    if response_data is None:
+        raise click.ClickException(response.text)
+
+    signing_roles: Dict[str, Any] = response_data.get("metadata", {})
+    if len(signing_roles) == 0:
+        raise click.ClickException("No metadata available for signing")
+
+    return signing_roles
+
+
+def _get_signing_key(role_info: MetadataInfo) -> RSTUFKey:
+    pending_keys: List = []
+    for key_id in role_info._new_md.signed.roles[Root.type].keyids:
+        if key_id not in role_info._new_md.signatures:
+            pending_keys.append(role_info._new_md.signed.keys[key_id])
+
+    sign_key_name = prompt.Prompt.ask(
+        "\nChoose a private key to load",
+        choices=[
+            signing_key.unrecognized_fields["name"]
+            for signing_key in pending_keys
+        ],
+    )
+
+    while True:
+        rstuf_key = _get_key(sign_key_name)
+        if rstuf_key.error:
+            console.print(rstuf_key.error)
+            retry = prompt.Confirm.ask(
+                f"\nRetry to load the key {sign_key_name}?"
+            )
+            if not retry:
+                raise click.ClickException("Aborted.")
+        else:
+            break
+
+    rstuf_key_id = rstuf_key.key["keyid"]
+
+    current_role_key = role_info._new_md.signed.keys[rstuf_key_id]
+    current_role_key_name = current_role_key.unrecognized_fields["name"]
+    if current_role_key_name != sign_key_name:
+        raise click.ClickException(f"Loaded key is not '{sign_key_name}'")
+
+    return rstuf_key
+
+
+def _sign_metadata(role_info: MetadataInfo, rstuf_key: RSTUFKey) -> Signature:
+    signer = role_info.get_signer(rstuf_key)
+    try:
+        signature = role_info._new_md.sign(signer)
+    except UnsignedMetadataError as err:
+        raise click.ClickException("Problem signing the metadata") from err
```

Comment:
Do we want to give better information on what could be the problem?
How will the user recover from this problem or the administrator?

Reply:
There is no much information from UnsignedMetadataError. It doesn't give us details. Nothing we can improve for now.